src/renderers/dom/client/eventPlugins/TapEventPlugin.js

```diff
@@ -20,19 +21,59 @@ var ViewportMetrics = require('ViewportMetrics');
 var isStartish = EventPluginUtils.isStartish;
 var isEndish = EventPluginUtils.isEndish;
 
+import type {TopLevelTypes} from 'EventConstants';
+import type {ReactInstance} from 'ReactInstanceType';
+import type {PluginModule} from 'PluginModuleType';
+
+/**
+ * We are extending the Flow 'Touch' declaration to enable using bracket
+ * notation to access properties.
+ * Without this adjustment Flow throws
+ * "Indexable signature not found in Touch".
+ * See https://github.com/facebook/flow/issues/1323
+ */
+type TouchPropertyKey =
+  'clientX' |
+  'clientY' |
+  'pageX' |
+  'pageY';
+
+declare class _Touch extends Touch {
+  [key: TouchPropertyKey]: number;
+}
+
+type AxisCoordinateData = {
+  page: TouchPropertyKey,
+  client: TouchPropertyKey,
+  envScroll: string,
```

Comment:
In a follow up diff would be great to type this as

```
'currentPageScrollLeft' | 'currentPageScrollTop'
```

along with adding `@flow` on `ViewportMetrics.js`: https://github.com/facebook/react/blob/67f8524e88abbf1ac0fd86d38a0477d11fbc7b3e/src/renderers/dom/client/utils/ViewportMetrics.js